src/tcgwars/logic/impl/gen4/RisingRivals.groovy

```diff
@@ -1454,17 +1922,43 @@ public enum RisingRivals implements LogicCardInfo {
           move "Collect", {
             text "Draw 2 cards."
             energyCost C
-            attackRequirement {}
+            attackRequirement {
+              assert my.deck : "Your deck is empty"
+            }
             onAttack {
-              damage 0
+              draw 2
             }
           }
           move "Primal Breath", {
             text "30 damage. Your opponent can’t play any Pokémon from his or her hand to Evolve or to Level Up the Defending Pokémon during his or her next turn."
             energyCost F, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 30
+              afterDamage {
+                targeted (defending) {
+                  bc "During ${opp.owner.getPlayerUsername(bg)}'s next turn, Energy can't be attached from their hand to the Defending ${defending}. (This effect can be removed by evolving or benching ${defending}.)"
```

Comment:
Text is wrong.

You're also missing several EVOLVE effects you need to handle https://github.com/axpendix/tcgone-engine-contrib/blob/b9adba8e4aa995503c40134400f67ce87db63210/src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy#L1322-L1347